core/server/data/validation/index.js

```diff
@@ -131,24 +128,17 @@ validateSettings = function validateSettings(defaultSettings, model) {
 };
 
 validateActiveTheme = function validateActiveTheme(themeName) {
-    // If Ghost is running and its availableThemes collection exists
-    // give it priority.
-    if (config.get('paths').availableThemes && Object.keys(config.get('paths').availableThemes).length > 0) {
-        availableThemes = Promise.resolve(config.get('paths').availableThemes);
+    // @TODO come up with something way better here - we should probably attempt to read the theme from the
+    // File system at this point and validate the theme using gscan rather than just checking if it's in a cache object
+    if (!config.get('paths').availableThemes || Object.keys(config.get('paths').availableThemes).length === 0) {
```

Comment:
Could you please explain, what is this condition protecting against?
And why do we resolve?